docs(graphql): Update mapping Graphql to Dgraph (#237)
dgraph-io/dgraph-docs: content/graphql/dgraph/index.md

If you have an existing Dgraph instance and want to also expose GraphQL, you need to add a GraphQL schema that maps to your Dgraph schema.  You don't need to expose your entire Dgraph schema as GraphQL, but do note that adding a GraphQL schema can alter the Dgraph schema.

> Dgraph also allows type and edge names that aren't valid in GraphQL, so, often, you'll need to expose valid GraphQL names. Dgraph admits special characters and even different languages (see [Predicate name rules]({{< relref "query-language/schema.md#predicate-name-rules">}})), while the GraphQL Spec requires that type and field (predicate) names are generated from `/[_A-Za-z][_0-9A-Za-z]*/`.

Reviewer: ```suggestion
Dgraph's native DQL syntax allows type and edge names that aren't valid in GraphQL; so, you'll often need to expose valid GraphQL names. Dgraph permits special characters, including Unicode characters from a variety of  languages (see [Predicate name rules]({{< relref "query-language/schema.md#predicate-name-rules">}})). Conversely, the [GraphQL specification on naming](https://spec.graphql.org/June2018/#sec-Names) requires that entity names, including types and fields (predicate), are composed of ASCII characters and generated as follows: `/[_A-Za-z][_0-9A-Za-z]*/`.
```

}
```

{{% notice "note" %}}

> The current behavior requires that when two fields are mapped to the same Dgraph predicate both should have the same `@search` directive.

Reviewer: ```suggestion
In Dgraph's current GraphQL implementation, if two fields are mapped to the same Dgraph predicate, both should have the same `@search` directive.
```

> This is likely to change in a future release where the underlying Dgraph indexes will be the union of the `@search` directives, while the generated GraphQL API will expose only the search given for the particular field.

Reviewer: I'd cut these two lines (98 and 99), as we already hint that this behavior might change, and detailing possible future implementations isn't a best practice.

Author: Lines removed.

> * Generating a first pass GraphQL schema from an existing dgraph schema.

Reviewer: I know it isn't 100% within the scope of this PR, but I'd be in favor of cutting the whole "## Roadmap" section... But maybe it makes more sense for me to file a task to remove "roadmap" /  detailed forward-looking statements in general. Not a merge blocker, just a thought.

Author: I removed the `Roadmap` section
